unstructured_ingest/v2/processes/connectors/couchbase.py

```diff
@@ -0,0 +1,159 @@
+import json
+from dataclasses import dataclass, field
+from datetime import timedelta
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Optional
+
+from unstructured_ingest.enhanced_dataclass import enhanced_field
+from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    UploadContent,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+)
+
+if TYPE_CHECKING:
+    from couchbase.cluster import Cluster
+
+CONNECTOR_TYPE = "couchbase"
+SERVER_API_VERSION = "1"
+
+
+@dataclass
+class CouchbaseAccessConfig(AccessConfig):
+    connection_string: Optional[str] = None
+    username: Optional[str] = None
+    password: Optional[str] = None
```

Comment:
I was thinking of something like this. Correct me if I'm wrong in what's required from this. 
```suggestion
    password: Optional[str] = None
    
    def __post_init__(self):
        if not self.connection_string and not (self.username and self.password):
            raise ValueError(f"Either connection string or username and password must be set")
```

Reply:
Actually what does the connection string look like? Would this contain the username/password info in it? I see in your code to get the connection, all three are required.

Reply:
Looking at this again, the only value that needs to be in the `access_config` is the password. Since neither the username nor connection string have sensitive data, these can live in the connection config.

Reply:
Also can a user connect to couchbase without a username or password? This will impact if these are required on the config and we might not need the checks in the `__post_init__` method.

Reply:
I have updated to only have password be sensitive.

> Also can a user connect to couchbase without a username or password?

No, username and password are separate from connection string and are required fields to connect to couchbase cluster